source/Ship.cpp

```diff
@@ -4531,7 +4531,8 @@ void Ship::StepTargeting()
 	shared_ptr<const Ship> target = GetTargetShip();
 	// If this is a fighter or drone and it is not assisting someone at the
 	// moment, its boarding target should be its parent ship.
-	if(CanBeCarried() && !(target && target == GetShipToAssist()))
+	// Unless the player uses a fighter as their flagship and is boarding an enemy ship.
+	if(CanBeCarried() && !(target && (target == GetShipToAssist() || isYours)))
```

Comment:
Will this not apply to all player owned ships, not just the flagship?

Reply:
Uhh.. It's been a minute since I worked on this. Iirc only your flagship can get to this point in the boarding process if it's a fighter?

I'll try to look into that again.

Reply:
I think the idea was that if it's your ship and is boarding an enemy, it must be your flagship?

Reply:
Also, I checked 4544, and the boarding status seems to be cleared for fighter flagships without touching it. It would be rather difficult to check if the ship is your flagship, though, so I'd rather leave that check as-is.